lib/trento.pm

```diff
@@ -63,12 +66,97 @@ use constant TRENTO_AZ_PREFIX => 'openqa-trento';
 # Azure does not support dash or underscore in ACR name
 use constant TRENTO_AZ_ACR_PREFIX => 'openqatrentoacr';
 use constant CYPRESS_IMAGE_TAG => 'goofy';
+use constant CYPRESS_IMAGE => 'docker.io/cypress/included';
 
 =head1 DESCRIPTION 
 
 Package with common methods and default or constant  values for Trento tests
 
 =head2 Methods
+=cut
+
+
+=head3 get_trento_deployment
+
+Get the set of scripts for the Trento deployment
+=cut
+
+sub get_trento_deployment {
+    my ($self, $work_dir) = @_;
+
+    enter_cmd "cd ${work_dir}";
+    my $gitlab_token = get_var(TRENTO_GITLAB_TOKEN => get_required_var('_SECRET_TRENTO_GITLAB_TOKEN'));
+
+    # Script from a release
+    if (get_var('TRENTO_DEPLOY_VER')) {
+
+        # Clean up whatever is present in the indicated working folder
+        # of the qcow2 image, to avoid confision
+        script_run('rm -rf *');
+
+        # Get some 'coordinates' from the test settings
+        my $ver = get_var('TRENTO_DEPLOY_VER');
+        my $gitlab_repo = get_var(TRENTO_GITLAB_REPO => 'gitlab.suse.de/qa-css/trento');
+        my @gitlab_url = split('/', $gitlab_repo);
+        my $gitlab_namespace = $gitlab_url[-2];
+        my $gitlab_project = $gitlab_url[-1];
+
+        my $curl_cmd = 'curl -s ' .
+          "--header \"PRIVATE-TOKEN: ${gitlab_token}\" ";
+        my $gitlab_api = 'https://gitlab.suse.de/api/v4/projects';
+
+        # Get the ID of the requested project
+        my $repo_id_cmd = $curl_cmd .
+          " \"${gitlab_api}/${gitlab_namespace}%2F${gitlab_project}\"" .
+          " | jq -r '.id'";
+        my $repo_id = script_output($repo_id_cmd);
+        # hotfix: harcoded here as not able to strip color codes added also if using -r
+        $repo_id = '7182';
+
+        my $repo_url = "${gitlab_api}/${repo_id}";
+        my $rel_api_url = "${repo_url}/releases/v${ver}";
+
+        # Get the file name of the release archive
+        my $ver_artifact_cmd = $curl_cmd .
+          " ${rel_api_url} " .
+          "| jq -r '.assets.sources[]|select(.format == \"tar.gz\")" .
+          "|.url'|rev|cut -d/ -f1|rev";
```

Comment:
Neat trick with the double `rev` commands, but you could do the same with a shorter shell pipeline and `File::Basename::basename` and even drop the `jq` and use `decode_json` as well.

If wanting to keep this in the shell and not in the Perl code, there is also `basename`:
```
$ echo /home/michelepa/trento/my_directory | rev | cut -d/ -f1 | rev
my_directory
$ basename /home/michelepa/trento/my_directory
my_directory
$ basename $(echo /home/michelepa/trento/my_directory)
my_directory
$ curl -s --header "PRIVATE-TOKEN: ****"  https://gitlab.url/api/v4/projects/7182/releases/v0.2.0 | jq -r '.assets.sources[]|select(.format == "tar.gz")|.url'|rev|cut -d/ -f1|rev
trento-v0.2.0.tar.gz
$ basename $(curl -s --header "PRIVATE-TOKEN: ****"  https://gitlab.url/api/v4/projects/7182/releases/v0.2.0 | jq -r '.assets.sources[]|select(.format == "tar.gz")|.url')
trento-v0.2.0.tar.gz
```